docker/Dockerfile

```diff
@@ -0,0 +1,182 @@
+#
+# Build:
+#   $ docker build docker -f docker/Dockerfile.tcf-dev -t avalon-shell-dev
+#   if behind a proxy, you might want to add also below options
+#   --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy --build-arg ftp_proxy=$ftp_proxy
+#
+# Run:
+#   $ cd avalon
+#   $ docker run -v $(pwd):/project/avalon -it avalon-shell-dev /bin/bash
+#   (to run with SGX HW, add options '--device=/dev/isgx -v /var/run/aesmd:/var/run/aesmd ')
+#   make clean && make
+
+
+FROM ubuntu:bionic as base_image
+
+# Ignore timezone prompt in apt
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Add necessary packages
+RUN apt-get update \
+ && apt-get install -y -q \
+    build-essential \
+    ca-certificates \
+    pkg-config \
+    python3-toml \
+    python3-requests \
+    python3-colorlog \
+    python3-twisted \
+    software-properties-common \
+ && apt-get -y -q upgrade \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Make Python3 default
+RUN ln -sf /usr/bin/python3 /usr/bin/python
+
+# Installing solidity compiler
+RUN add-apt-repository ppa:ethereum/ethereum \
+   && apt-get update \
+   && apt-get install solc
+
+#Build Avalon Enclave Manager intermediate docker image
+FROM base_image as build_image
+
+RUN apt-get update \
+ && apt-get install -y -q \
+    cmake \
+    make \
+    libprotobuf-dev \
+    python3-dev \
+    python3-pip \
+    swig \
+    software-properties-common \
+    wget \
+    tar
+
+# Install setuptools,py-solc-x and web3 packages using pip because
+# these are not available in apt repository.
+# Install nose2 for running tests.
+RUN pip3 install --upgrade setuptools json-rpc web3 py-solc-x nose2
+
+WORKDIR /tmp
+# Build ("Untrusted") OpenSSL, then build ("trusted") SGX OpenSSL
+# Note: The latter will compile in HW or SIM mode depending on the
+# availability of /dev/isgx and /var/run/aesmd/aesm.socket
+RUN OPENSSL_VER=1.1.1d \
+ && wget https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz \
+ && tar -zxf openssl-$OPENSSL_VER.tar.gz \
+ && cd openssl-$OPENSSL_VER/ \
+ && ./config \
+ && THREADS=8 \
+ && make -j$THREADS \
+ && make test \
+ && make install -j$THREADS \
+ && ldconfig \
+ && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/ \
+ && cd .. \
+ && rm -rf openssl-$OPENSSL_VER
+
+COPY VERSION /project/avalon/
+COPY ./bin /project/avalon/bin
+COPY ./tools/run_tests.sh /project/avalon/tools/
+COPY ./config/tcs_config.toml /project/avalon/config/
+COPY ./common/cpp /project/avalon/common/cpp
+COPY ./common/python /project/avalon/common/python
+COPY ./sdk /project/avalon/sdk
+COPY ./tests /project/avalon/tests
+COPY ./examples/apps /project/avalon/examples/apps
+
+WORKDIR /project/avalon/common/cpp
+
+RUN echo "Building Avalon Common CPP\n" \
+  && mkdir -p build \
+  && cd build \
+  && cmake .. -DUNTRUSTED_ONLY=1 \
+  && make
+
+WORKDIR /project/avalon/common/python
+
+RUN echo "Building Avalon Common Python\n" \
+  && make \
+  && make install
```

Comment:
installing wheel file (via make install) should be done in final image. Refer shared_kv_storage Dockerfile. https://github.com/hyperledger/avalon/blob/master/shared_kv_storage/Dockerfile

Reply:
I referred this from listener dockerfile. I will install the wheel package in the final image.